tests/unit/optimiser/test_team_selector.py

```diff
@@ -0,0 +1,489 @@
+        "key": 6,
+        "value": 40,
+        "element_type": 4,
+        "team": 17,
+        "safe_web_name": "quinn",
+    },
+    {
+        "element": 18,
+        "key": 6.5,
+        "value": 40,
+        "element_type": 4,
+        "team": 18,
+        "safe_web_name": "rich",
+    },
+    {
+        "element": 19,
+        "key": 7,
+        "value": 40,
+        "element_type": 4,
+        "team": 19,
+        "safe_web_name": "sid",
+    },
+    {
+        "element": 20,
+        "key": 3,
+        "value": 40,
+        "element_type": 4,
+        "team": 20,
+        "safe_web_name": "tom",
+    },
+]
+
+players = keepers + defenders + midfielders + forwards
+
+
+def test_construct_player_position_matrix():
+    m = np.array(
+        [
+            [
+                1.0,
+                1.0,
+                1.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+            ],
+            [
+                0.0,
+                0.0,
+                0.0,
+                1.0,
+                1.0,
+                1.0,
+                1.0,
+                1.0,
+                1.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+            ],
+            [
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                1.0,
+                1.0,
+                1.0,
+                1.0,
+                1.0,
+                1.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+            ],
+            [
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                1.0,
+                1.0,
+                1.0,
+                1.0,
+                1.0,
+            ],
+        ]
+    )
+
+    assert np.array_equal(construct_player_position_weights(players), m)
+
+
+def test_construct_player_team_matrix():
+    m = np.identity(20)
+
+    assert np.array_equal(construct_player_team_weights(players), m)
+
+
+def test_select_team_from_scratch():
+
+    first_team, bench, captain, vice, transfers = select_team(
```

Comment:
we could use a [namedtuple](https://docs.python.org/3/library/collections.html#collections.namedtuple) to reduce risk of unpacking the return value to wrong-named variable- if you mix up the ordering here would be confusing

Reply:
Looks like a neat trick, but I think it's okay for the time being.